Dockerfile

```
@@ -491,45 +453,115 @@ ENV CGO_CFLAGS="${CFLAGS}"
 ENV CGO_CXXFLAGS="${CXXFLAGS}"
 ENV CGO_LDFLAGS="${LDFLAGS}"
 
+ENV GO111MODULE="auto"
+
+ENV AWS_LC_FIPS_VER="2.0.9"
+
+USER root
+RUN dnf -y install golang
+
+ENV GO121VER="1.21.9"
+ENV GO122VER="1.22.1"
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-go-prep as sdk-go-1.21-prep
+
+USER builder
+
+WORKDIR /home/builder/sdk-go-${GO121VER}
+
+COPY ./hashes/go-1.21 /home/builder/hashes-go
+COPY ./helpers/go/* ./
+COPY ./patches/go-1.21 /home/builder/patches-go
+
+COPY ./hashes/aws-lc /home/builder/hashes-aws-lc
+COPY ./patches/aws-lc /home/builder/patches-aws-lc
+
+RUN ./prep-go.sh --go-version=${GO121VER} 
+
+WORKDIR /home/builder/aws-lc/build
+COPY ./configs/aws-lc/* .
+COPY ./helpers/aws-lc/*   .
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-go-prep as sdk-go-1.22-prep
+
+USER builder
+
+WORKDIR /home/builder/sdk-go-${GO122VER}
+
+COPY ./hashes/go-1.22 /home/builder/hashes-go
+COPY ./helpers/go/* ./
+COPY ./patches/go-1.22 /home/builder/patches-go
+
+COPY ./hashes/aws-lc /home/builder/hashes-aws-lc
+COPY ./patches/aws-lc /home/builder/patches-aws-lc
+
+RUN ./prep-go.sh --go-version=${GO122VER}
+
 WORKDIR /home/builder/aws-lc/build
 COPY ./configs/aws-lc/* .
 COPY ./helpers/aws-lc/* .
 
 # =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
-FROM sdk-go-prep as sdk-go-aws-lc-x86_64
+FROM sdk-go-1.21-prep as sdk-go-1.21-aws-lc-x86_64
+ENV ARCH="x86_64"
+RUN ./build-aws-lc.sh --arch="${ARCH}" --go-dir="${HOME}/sdk-go-${GO121VER}"
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-go-1.21-prep as sdk-go-1.21-aws-lc-aarch64
+ENV ARCH="aarch64"
+RUN ./build-aws-lc.sh --arch="${ARCH}" --go-dir="${HOME}/sdk-go-${GO121VER}"
+
+# =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+FROM sdk-go-1.22-prep as sdk-go-1.22-aws-lc-x86_64
 ENV ARCH="x86_64"
-RUN ./build-aws-lc.sh --arch="${ARCH}" --go-dir="${HOME}/sdk-go"
+RUN ./build-aws-lc.sh --arch="${ARCH}" --go-dir="${HOME}/sdk-go-${GO122VER}"
 
 # =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
-FROM sdk-go-prep as sdk-go-aws-lc-aarch64
+FROM sdk-go-1.22-prep as sdk-go-1.22-aws-lc-aarch64
 ENV ARCH="aarch64"
-RUN ./build-aws-lc.sh --arch="${ARCH}" --go-dir="${HOME}/sdk-go"
+RUN ./build-aws-lc.sh --arch="${ARCH}" --go-dir="${HOME}/sdk-go-${GO122VER}"
 
 # =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
-FROM sdk-go-prep as sdk-go
+FROM sdk-go-1.21-prep as sdk-go-1.21
 
-COPY --from=sdk-go-aws-lc-x86_64 \
+COPY --from=sdk-go-1.21-aws-lc-x86_64 \
   /home/builder/aws-lc/build/goboringcrypto_linux_amd64.syso \
-  /home/builder/sdk-go/src/crypto/internal/boring/syso/goboringcrypto_linux_amd64.syso
+  /home/builder/sdk-go-${GO121VER}/src/crypto/internal/boring/syso/goboringcrypto_linux_amd64.syso
```

This is where removing the go version from the path starts to pay off - because then these `COPY` commands can just be copy-pasted into new stages:
```suggestion
  /home/builder/sdk-go/src/crypto/internal/boring/syso/goboringcrypto_linux_amd64.syso
```

---

nice - much cleaner :)